main.go

```diff
@@ -141,10 +146,15 @@ func main() {
 		LeaseDuration:                 &leaderElectionOptions.LeaseDuration,
 		RenewDeadline:                 &leaderElectionOptions.RenewDeadline,
 		RetryPeriod:                   &leaderElectionOptions.RetryPeriod,
-		LeaderElectionID:              fmt.Sprintf("%s-leader-election", controllerName),
+		LeaderElectionID:              leaderelection.GenerateID(controllerName, watchOptions.LabelSelector),
```

Comment:
You need to check for an `LabelSelector != ""` and compose it until I have fixed `GenerateID` to deal with empty string arguments, as otherwise it will be suffixed with `-e3b0c4429`.

Reply:
Example: https://github.com/fluxcd/source-controller/blob/52d7bab10d3744bc9631c158f9f23be6fc224382/main.go#L335-L338